docs/testing/e2e-framework-design-dimensions.md

```diff
@@ -0,0 +1,396 @@
+
+
+<a id="org9d1dbe8"></a>
+
+#### Possible Operations
+
+-   Boolean comparison operators
+-   Set membership operators
+
+
+<a id="orgecd6ea6"></a>
+
+## Value Domains
+
+Two value domains are recognized at this time:
+
+-   **Discreet:** A set of distinct and countable values. For example, years of schooling, or computer architectures.
+-   **Continuous:** Any of an infinite number of values within a range. For example, a timed duration, or a file size.
+
+At first glance, the complications involved in supporting continuous values can seem daunting. In practice howerver, the number of distinct values encountered are generally on the order of 2–100 and so for the purposes of an end-to-end test can usually be handled like discreet values. As an example, a duration has an infinite number of possible values, but for the purposes of an end-to-end test they can be rounded to a coarser-grained precision, effectively mapping them to discreet values. In other cases, like a semantic version, the number of possible values is infinite, but the number of supported values is likely much smaller, and discreet.
+
+> It's not clear if value domains will have any impact on the test framework's implementation, but it could be useful in reasoning about or implementing comparison operations.
+
+
+<a id="org666546a"></a>
+
+## Recognized Test Dimensions
+
+These dimensions have been identified as having an impact on end-to-end tests, and are therefore candidates for implementation in the system.
+
+
+<a id="org65dd08b"></a>
+
+### Provider
+
+|                      |          |
+|----------------------|----------|
+| Type                 | string   |
+| Domain               | discreet |
+| Cardinality estimate | < 10     |
+
+
+Values:
+
+-   Bare Metal
+-   CloudStack
+-   Docker
+-   Snowball
+-   vSphere
+
+
+<a id="org0df1bd1"></a>
+
+### Provider Version
+
+|                      |                                                        |
+|----------------------|--------------------------------------------------------|
+| Type                 | string                                                 |
+| Domain               | continuous                                             |
+| Cardinality estimate | < 5 per provider ||
+| Note                 | The values are likely to be similar to semver.         |
+
+> Are different provider versions supported? If so, which ones? How will we test on different versions of vSphere? What constitutes a different version of Bare Metal? Does a version even mean anything when running EKS-A on CloudStack?
+
+| Provider   | Versions Supported         |
+|------------|----------------------------|
+| Bare Metal | N/A                        |
+| CloudStack | 4.14+                      |
+| Docker     | ? (TBD)                    |
+| Snowball   | all (which is 1 right now) |
+| vSphere    | v7+ (major releases only?) |
+
+
+<a id="org1f576cb"></a>
+
+### CLI Version
```

Comment:
Should the CLI even be part of this framework? It would make sense for CLI test to remain as is in the eks-anywhere and those should just handle the package installation